### PR TITLE
Add fuzzer for HttpMetaFetcher

### PR DESCRIPTION
**Pull request by arthurscchan**
#### Summary

Add fuzzer for dev.sigstore.tuf.HttpMetaFetcher class.

### CODE REVIEW

**Comment by arthurscchan** [comment]
@AdamKorcz

File: fuzzing/build.gradle.kts [hunk 1]
```diff
@@ -8,7 +8,17 @@ repositories {
 
 dependencies {
     implementation(project(":sigstore-java"))
+
+    implementation(platform("io.grpc:grpc-bom:1.46.0"))
```
Comment (loosebazooka) [inline fuzzing/build.gradle.kts:12]: same here, limiting the dependencies to what is used.

Comment (arthurscchan) [reply fuzzing/build.gradle.kts]: Thanks for the suggestion. Have removed unused dependencies.

**Comment by loosebazooka** [comment]
closing for now, we can increase fuzzing coverage at another time. These have been open for too long.

**Pull request closed without merging**